File: rust/src/problems/p70.rs
pub struct Solution;

impl Solution {
    pub fn climb_stairs(n: i32) -> i32 {
        if n <= 1 {
            return 1;
        }

        let mut prev_1 = 1;
        let mut prev_2 = 1;
        for _ in 2..=n {
            let t = prev_1 + prev_2;
            prev_2 = prev_1;
            prev_1 = t;
        }
        prev_1
    }
}

#[cfg(test)]
mod tests {
    use super::*;

    #[test]
    fn case1() {
        assert_eq!(2, Solution::climb_stairs(2));
    }

    #[test]
    fn case2() {
        assert_eq!(3, Solution::climb_stairs(3));
    }
}

File: cpp/p70.cpp
#include <cassert>

using namespace std;

class Solution {
public:
    int climbStairs(int n) {
        int f0 = 1, f1 = 1;
        for (int i = 2; i <= n; i++) {
            int t = f1;
            f1 += f0;
            f0 = t;
        }
        return f1;
    }
};

int main() {
    assert(3, Solution().climbStairs(3));
    assert(1836311903, Solution().climbStairs(45));
    return 0;
}